apps/daimo-mobile/src/common/nav.ts

```diff
@@ -214,13 +214,44 @@ export async function handleOnboardingDeepLink(
   nav: OnboardingNav,
   str: string
 ) {
-  const inviteLink = parseInviteCodeOrLink(str);
   console.log(`[INTRO] paste invite link: '${str}'`);
+
+  const link = parseDaimoLink(str);
+  if (link == null) {
+    console.log(`[INTRO] skipping unparseable link ${str}`);
+    nav.navigate("CreateNew");
+    return;
+  }
+  if (link.type !== "invite" && link.type !== "tag") {
+    console.log(`[INTRO] skipping non-onboarding deep link ${str}`);
+    nav.navigate("CreateNew");
+    return;
+  }
+
+  const linkStatus = await fetchLinkStatus(link, dc);
+  if (linkStatus.link.type !== "invite") {
+    console.log(
+      `[INTRO] got link status of non-invite link ${str}. status: ${JSON.stringify(
+        linkStatus.link
+      )} is not an invite link`
+    );
+    nav.navigate("CreateNew");
+    return;
+  }
+
+  const inviteStatus = linkStatus as DaimoInviteCodeStatus;
   const isAndroid = Platform.OS === "android";
-  if (inviteLink && (await fetchInviteLinkStatus(dc, inviteLink))?.isValid) {
```

Comment:
i like the simplicity of the previous way here.

also, i think we can fix this server-side too!

fetchInviteLinkStatus does two things:
- call `fetchLinkStatus` via RPC
- send the result thru `getInviteStatus` locally

...we can fix `fetchLinkStatus` so that, when given a tag URL, it returns

```
{
   link: { /* the resolved link */ },
   ...
}
```

Reply:
hmm... would be good to move more of the logic server-side.

however, the `fetchInviteLinkStatus` code isn't being called because this line parses tags as `undefined`, so i think we need to do an app release either way
```
const inviteLink = parseInviteCodeOrLink(str);
```

Reply:
makes sense.
yeah, i think this section in parseInviteCodeOrLink could be simplified

before:
```ts
const link = parseDaimoLink(str);
  if (link != null && ["invite", "requestv2", "notev2"].includes(link.type)) {
    return link;
  }

return undefined;
```
  
after:

```ts
return parseDaimoLink(str);
```